### ISSUE TITLE
Feature request: LV2 version?

### ISSUE THREAD

**Issue by mxmilkiib** [mzuther/Squeezer#12]
### Is your feature request related to a problem? If so, please describe.

LV2 offers better features and is the de facto libre audio plugin format.

### Describe the solution you'd like

There is a fork of JUCE that can target LV2, https://github.com/DISTRHO/juce/tree/juce6, with @falktx and @kottv being the main drivers of the project.

**Comment by mzuther** [comment]
Hi!

Thanks for your feature request. You can download a LV2 version from here: https://github.com/mzuther/Squeezer/releases/tag/v2.5.3

However, there will be no newer LV2 releases because falkTX has stopped LV2 development, there are few applications supporting it and the LV2 plugin took a lot of effort in maintaining. In addition, I think that VST3 will soon become the new de facto libre audio plugin format, now that it is GPL'ed.

Best regards,

Martin

**Comment by falkTX** [comment]
> [](https://github.com/mzuther/Squeezer/releases/tag/v2.5.3)
> 
> However, there will be no newer LV2 releases because falkTX has stopped LV2 development

This is simply not true. I had my beef with JUCE for very good reasons, but cant deny it is useful.
The DISTRHO JUCE fork has been updated to build with latest JUCE v6.

Doing LV2 with JUCE is difficult mostly because the official devs make it so. It is very sad that they outright refuse to give support for it, even with all the work already done for them.

In any case, I have my doubts VST3 will succeed in becoming the real go-to format for libre audio things.
It barely runs on Linux, even less so on others like BSD. LV2 has no problems at all working on pretty much all platforms.
Personally really dislike how VST3 works, they make you build against this codebase that is full of bugs and ugly compiler warnings, lead by a company interested on profit - contrary to LV2 which is a community project where public discussion takes place.
VST3 will get more popular because of commercial DAWs and plugins, so not much because of open-source ones.

**Comment by falkTX** [comment]
On a somewhat related note, would you mind if I add Squeezer to DISTRHO-Ports? People are interested on LV2 support, which I can easily maintain separately in there.

**Comment by mzuther** [comment]
@falkTX I remember reading a message of yours that said you ditched LV2 support for JUCE. I'm sorry if I got you wrong. In any case, it's great to know that LV2 is back! :smile:

I'll re-open this issue and think of bringing LV2 support back. And of course you can add Squeezer to DISTRHO, I'm flattered! However, I have not yet ported it to JUCE v6 yet because I wanted most bugs to be fixed first. I've had my problems with JUCE's code base before...

(I cannot see into the future, but *if* JUCE supports Linux VST3 plug-ins well, that might help it gain some traction. However, I'm not a big fan of VST3 myself. Well, the future will tell…)

**Comment by falkTX** [comment]
Thanks for reconsidering, that is good to know.

And yes, I was going to give up LV2 support in JUCE completely. But as I hope to see LV2 succeed, I feel I can't just leave things like that.
So anyway, I will continue to work on the DISTRHO/JUCE fork

One thing worth noting is that because the DISTRHO-Ports has quite some old plugins that are basically in maintenance mode (I am not the author, I just fix some small issues with them where possible), I need to keep an older version of JUCE around.
So for the DISTRHO-Ports project, it actually builds the new and old juce. Linux-VST3 is only possible with the new JUCE, but LV2 is possible on both.

**Comment by mzuther** [comment]
@falkTX I have moved to JUCE 6 and am about to resurrect the LV2 versions. In the past days, I have dabbled with dynamic I/O allocation in VST3. Do LV2 and DISTRO/JUCE support this as well or do I need to compile distinct versions for mono and stereo?

Merry Christmas! :smile:

**Comment by falkTX** [comment]
You need 2 plugin variants, as before.
There is no LV2 extension for dynamic IO at this point.

**Comment by mzuther** [comment]
Thanks, that has probably saved me an hour or so.

I have compiled a 64-bit version: https://public.mzuther.de/?f37c68e938b8f20c69db6d379605bec748229f76

Could you one of you please try it when you find the time? Thanks!

**Comment by falkTX** [comment]
how do I build these?

**Comment by mzuther** [comment]
The link contains the binaries – to build yourself, simply run `Builds/build.sh`.

By the way, where can I get / compile an up-to-date version of `lv2_ttl_generator`? Google wouldn't find it...

**Comment by mzuther** [comment]
I forgot to mention that there is a whole section on compiling in the manual (directory `doc`). It might be slightly out of date, though.

**Comment by rghvdberg** [comment]
Maybe look in DPF for the ttl generator? https://github.com/DISTRHO/DPF/tree/master/utils/lv2-ttl-generator

**Comment by mzuther** [comment]
Thanks! :smile:

**Comment by mzuther** [comment]
I have tried to get the LV2 plug-ins to run and it seems that JUCE's new multibus API is not supported. However, I rely on it to make VST3 work as it should. As my time is limited, I have decided to officially drop support for LV2. I'm really sorry, it detracts me from more important issues (such as implementing oversampling).

@falkTX You are more than welcome to create LV2 plug-ins for DISTRHO! For the time being, I'll leave the infrastructure in place – you just have to add a LV2 section to `Builds/settings.jinja` like [this](https://github.com/mzuther/Squeezer/blob/38896e185583ad41abf3768578d013c5084a986c/Builds/settings.jinja#L40). You are also welcome to open PRs if you see the need!

Thanks to all of you and have a great new year!
